packages/myst-cli/src/transforms/outputs.ts

```diff
@@ -47,6 +48,99 @@ export async function transformOutputsToCache(
   );
 }
 
+export function stringIsMatplotlibOutput(value?: string): boolean {
+  if (!value) return false;
+  // We can add more when we find them...
+  const match =
+    value.match(/<(Figure|Text|matplotlib)(.*)at ([0-9a-z]+)>/) ||
+    value.match(/^<(Figure|Text|AxesSubplot)(.*)>$/) ||
+    value.match(/^(Text)\((.*)\)$/);
+  return !!match;
+}
+
+/** Remove warnings and errors from outputs */
+export function transformFilterOutputStreams(
+  mdast: GenericParent,
+  vfile: VFile,
+  {
+    output_stdout: stdout = 'show',
+    output_stderr: stderr = 'show',
+    output_matplotlib_strings: mpl = 'remove-warn',
+  }: Pick<ProjectSettings, 'output_stdout' | 'output_stderr' | 'output_matplotlib_strings'> = {},
+) {
+  if (stdout === 'show' && stderr === 'show') return;
+  const blocks = selectAll('block', mdast) as GenericNode[];
+  blocks.forEach((block) => {
+    const tags: string[] = Array.isArray(block.data.tags) ? block.data.tags : [];
+    const blockRemoveStderr = tags.includes('remove-stderr');
+    const blockRemoveStdout = tags.includes('remove-stdout');
+    const outputs = selectAll('output', block) as GenericNode[];
+    // There should be only one output in the block
+    outputs.forEach((output) => {
+      output.data = output.data.filter((data: IStream | MinifiedMimeOutput) => {
+        if (stderr !== 'show' && data.output_type === 'stream' && data.name === 'stderr') {
```

Comment:
Should this also capture `output_type === 'error'`?

Reply:
I think we will want to do errors/exceptions in a new pass?